src/coordination/include/coordination/utils.hpp

```diff
@@ -0,0 +1,101 @@
+// Copyright 2024 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+#pragma once
+
+#include <optional>
+#include <range/v3/view.hpp>
+#include <ranges>
+#include <string>
+#include <unordered_map>
+#include "dbms/constants.hpp"
+#include "replication_coordination_glue/common.hpp"
+
+namespace memgraph::coordination {
+
+inline std::string ChooseMostUpToDateInstance(
+    const std::vector<std::pair<std::string, replication_coordination_glue::DatabaseHistories>>
+        &instance_database_histories,
+    std::optional<std::string> &latest_epoch, std::optional<uint64_t> &latest_commit_timestamp) {
+  std::string most_up_to_date_instance;
+  std::for_each(
+      instance_database_histories.begin(), instance_database_histories.end(),
+      [&latest_epoch, &latest_commit_timestamp, &most_up_to_date_instance](
+          const std::pair<const std::string, replication_coordination_glue::DatabaseHistories> &instance_res_pair) {
+        const auto &[instance_name, instance_db_histories] = instance_res_pair;
+
+        // Find default db for instance and its history
+        auto default_db_history_data =
+            std::find_if(instance_db_histories.begin(), instance_db_histories.end(),
+                         [default_db = memgraph::dbms::kDefaultDB](
+                             const replication_coordination_glue::DatabaseHistory &db_timestamps) {
+                           return db_timestamps.name == default_db;
+                         });
+
+        std::for_each(
+            instance_db_histories.begin(), instance_db_histories.end(),
+            [instance_name = instance_name](const replication_coordination_glue::DatabaseHistory &db_history) {
+              spdlog::trace("Instance {}: name {}, default db {}", instance_name, db_history.name,
+                            memgraph::dbms::kDefaultDB);
+            });
```

Comment:
delete this logging

Reply:
This is one point for discussion, I can actually reduce this whole code if I just return last_commit_timestamp from each database and find one that has biggest last commit timestamp. But if something goes sideways, we will have no idea where it went sideways. With users setting log level to DEBUG or TRACE, we can see exactly where we had issue really fast. I would argue as we have experimental flag, this could benefit us in short run, I can afterwards remove it, what do you think

Reply:
On call: change to DEBUG mode log level